src/redux/reducers/searchText.js

```diff
@@ -0,0 +1,198 @@
+import { handleActions } from 'redux-actions';
+import actionTypes from '../actionTypes';
+import { searchMessages } from '../../utils/searchMessages';
+
+const defaultState = {
+  search: false,
+  searchState: {
+    markedMessagesByID: {},
+    markedIDs: [],
+    markedIndices: [],
+    scrollLimit: 0,
+    scrollIndex: -1,
+    scrollID: null,
+    caseSensitive: false,
+    searchIndex: 0,
+    searchText: '',
+  },
+};
+
+export default handleActions(
+  {
+    [actionTypes.SEARCH_TEXT_CHANGED](state, { payload }) {
+      // TODO: Call close snackbar from Top Bar
+      const { messages, messagesByID, searchText } = payload;
+      let { searchState } = state;
+      // const searchText = event.target.value;
+      if (searchText.trim()) {
```

Comment:
We should consider that the searchText received in the reducer is already trimmed